apps/api/src/services/competition-manager.service.ts

```diff
@@ -973,20 +1006,23 @@ export class CompetitionManager {
       throw error;
     }
 
-    // 5. Check agent status is eligible
+    // 5. Check participant limit
+    await this.validateParticipantLimit(competition);
```

Comment:
this is racy.  If two agents try to register at the same time, they could both get past this check and be added, even if there's only room left for one of them.  This will get even worse when we have multiple app servers processing requests and updating the database in parallel.

This check should be done within the same database transaction that adds the user to the competition, so that the transaction succeeds or fails atomically including the agent limit check